### ISSUE TITLE
caching simulations

### ISSUE THREAD

**Issue by connor-french** [role-model/roleR#29]
@ajrominger and I were brainstorming the most memory-efficient way to cache simulations for roleShiny and had a few thoughts about what needs to be output/what is already output. The problem is that roleR simulation objects are huge (100s of megabytes) and caching more than a couple would become unwieldy.

For minimal memory overhead for caching simulations, only output the parameters, a unique seed per sim, and whatever processed output is needed for roleShiny do what it needs (this last point may be unique to roleShiny).
* low memory usage
* problem- want final state for initiating other sims

What may be helpful for thinking through what we need- enumerate everything that's being output with model. What is necessary and what is fluff?

Some ideas about why the roleR objects are so huge:
* phylogeny may be taking up a bunch of space
* think about how the edge matrix is being populated
* could be faster to just augment this matrix

**Comment by jidec** [comment]
Yup I think that changing how we augment matrices and deal with the empty spots in augmented data will make a huge difference here. For example once the model is done we can chop off unused rows of each matrix. And also in the beginning augment the phylogeny based on how strong the speciation parameter is rather than a flat amount every time.

**Comment by ajrominger** [comment]
agreed to those two great ideas:
1. chop off unused rows 
2. buffer the matrices by an amount the reflects the speciation 

we might even want to profile a version that has no buffer, but rather concatenates onto the matrices as needed (this could be faster is those concatenation events are rare)
